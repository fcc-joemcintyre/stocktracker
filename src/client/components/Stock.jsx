/**
 * Copyright (c) Joe McIntyre, 2016
 * license: MIT (https://github.com/fcc-joemcintyre/stocktracker/LICENSE.txt)
 */
import React from 'react';

export default class Stock extends React.Component {
  constructor (props) {
    super (props);
    this.onRemove = this.onRemove.bind (this);
  }

  onRemove () {
    console.log ('onRemove', this.props.symbol);
    this.props.removeStock (this.props.symbol);
  }

  render() {
    return (
      <div className='stockBox'>
        <div className='stockColor' style={{backgroundColor:this.props.color}}></div>
        <div className='stockBody'>
          <div className='stockSymbol'>{this.props.symbol}</div>
          <div className='stockName'>{this.props.name}</div>
        </div>
        <div className='stockRemove' onClick={this.onRemove}>X</div>
      </div>
    );
  }
}
