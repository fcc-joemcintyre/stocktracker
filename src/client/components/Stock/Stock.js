import React, { Component } from 'react';
import PropTypes from 'prop-types';

export class Stock extends Component {
  constructor (props) {
    super (props);
    this.onRemove = this.onRemove.bind (this);
  }

  onRemove () {
    this.props.onRemoveStock (this.props.symbol);
  }

  render () {
    return (
      <div className='stockBox'>
        <div className='stockColor' style={{ backgroundColor: this.props.color }} />
        <div className='stockBody'>
          <div className='stockSymbol'>{this.props.symbol}</div>
          <div className='stockName'>{this.props.name}</div>
        </div>
        <div className='stockRemove' onClick={this.onRemove}>X</div>
      </div>
    );
  }
}

Stock.propTypes = {
  name: PropTypes.string.isRequired,
  symbol: PropTypes.string.isRequired,
  color: PropTypes.string.isRequired,
  onRemoveStock: PropTypes.func.isRequired,
};
