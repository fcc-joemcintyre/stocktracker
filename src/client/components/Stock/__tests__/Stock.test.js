/* eslint-env jest */
import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import { Stock } from '../Stock';

test ('Stock displays its content', () => {
  const component = renderer.create (
    <Stock name='General Electric' symbol='GE' color='#0000ff' onRemoveStock={() => { /* */ }} />
  );
  const tree = component.toJSON ();
  expect (tree).toMatchSnapshot ();
});

test ('Remove stock event', () => {
  const onRemove = jest.fn ();
  const component = mount (
    <Stock name='General Electric' symbol='GE' color='#0000ff' onRemoveStock={onRemove} />
  );
  expect (component.props ().onRemoveStock).toBeDefined ();

  const button = component.find ('.stockRemove').first ();
  expect (button).toBeDefined ();

  button.simulate ('click');
  expect (onRemove).toBeCalledWith ('GE');
});
