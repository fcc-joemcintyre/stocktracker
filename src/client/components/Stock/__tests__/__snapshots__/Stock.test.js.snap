// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stock displays its content 1`] = `
<div
  style={
    Object {
      "border": "1px solid darkgray",
      "display": "flex",
      "height": "68px",
      "margin": "8px 8px 16px 0",
      "width": "200px",
    }
  }
>
  <div
    style={
      Object {
        "backgroundColor": "#0000ff",
        "height": "100%",
        "width": "10px",
      }
    }
  />
  <div
    style={
      Object {
        "height": "100%",
        "padding": "4px",
        "width": "100%",
      }
    }
  >
    <div
      style={
        Object {
          "fontSize": "24px",
        }
      }
    >
      GE
    </div>
    <div
      style={
        Object {
          "fontSize": "16px",
        }
      }
    >
      General Electric
    </div>
  </div>
  <div
    id="stock-remove"
    onClick={[Function]}
    style={
      Object {
        "color": "red",
        "cursor": "pointer",
        "textAlign": "center",
      }
    }
  >
    X
  </div>
</div>
`;
